src/main/scala/swiss/dasch/domain/Asset.scala

```diff
@@ -4,24 +4,70 @@ import eu.timepit.refined.api.Refined
 import eu.timepit.refined.types.string.NonEmptyString
 import swiss.dasch.config.Configuration.StorageConfig
 import zio.*
+import zio.json.{ DeriveJsonDecoder, DeriveJsonEncoder, JsonDecoder, JsonEncoder }
 import zio.nio.file.{ Files, Path }
 
 import java.io.IOException
 
 opaque type ProjectShortcode = NonEmptyString
 type IiifPrefix              = ProjectShortcode
 
-trait AssetService {
-  def listAllProjects(): IO[IOException, Chunk[String]]
+object ProjectShortcode {
+  def make(shortcode: String): Either[String, ProjectShortcode] = NonEmptyString.from(shortcode)
+  def unsafeFrom(shortCode: String): ProjectShortcode           = NonEmptyString.unsafeFrom(shortCode)
+}
+
+final case class DotInfoFileContent(
+    internalFilename: String,
+    originalInternalFilename: String,
+    originalFilename: String,
+    checksumOriginal: String,
+    checksumDerivative: String,
+  )
+
+object DotInfoFileContent {
+  implicit val decoder: JsonDecoder[DotInfoFileContent] = DeriveJsonDecoder.gen[DotInfoFileContent]
+  implicit val encoder: JsonEncoder[DotInfoFileContent] = DeriveJsonEncoder.gen[DotInfoFileContent]
```

Comment:
when you have a decoder and an encoder, you can simplify that to a codec. Something along those lines:
```suggestion
  implicit val codec: JsonCodec[DotInfoFileContent] = DeriveJsonCodec.gen[DotInfoFileContent]
```

Reply:
Thanks for the hint!